test/unit/concerns/cockpit_test.rb

```diff
@@ -0,0 +1,16 @@
+require 'test_helper'
+
+class DummyCockpit
+  include Cockpit
+end
+
+class CockpitTest < ActiveSupport::TestCase
+  test 'cockpit is enabled if cockpit ping is successful' do
+    dummy_cockpit = DummyCockpit.new
+    dummy_cockpit.stubs(:operatingsystem).returns(OpenStruct.new(:type => 'Fedora'))
```

Comment:
OpenStruct is a bit slow, if you want to suggest another altneratives that'd be great.